lib/spack/spack/test/web.py

```diff
@@ -210,16 +211,16 @@ def test_list_url(tmpdir):
     testpath = str(tmpdir)
     testpath_url = url_util.path_to_file_url(testpath)
 
-    os.mkdir(os.path.join(testpath, "dir"))
+    Path(testpath, "dir").mkdir()
 
-    with open(os.path.join(testpath, "file-0.txt"), "w"):
+    with Path(testpath, "file-0.txt").open("w", encoding="utf-8"):
         pass
-    with open(os.path.join(testpath, "file-1.txt"), "w"):
+    with Path(testpath, "file-1.txt").open("w", encoding="utf-8"):
         pass
-    with open(os.path.join(testpath, "file-2.txt"), "w"):
+    with Path(testpath, "file-2.txt").open("w", encoding="utf-8"):
         pass
 
-    with open(os.path.join(testpath, "dir", "another-file.txt"), "w"):
+    with Path(testpath, "dir", "another-file.txt").open("w", encoding="utf-8"):
```

Comment:
Is the encoding part required? IIRC the `os` method (like the rest of Python) assumes `utf-8`, does Pathlib differ?

Reply:
Keep the encoding, we're making them all "utf-8" now by default anyway.